### PR TITLE
Project: City Weather Web Application

### PR DESCRIPTION
**Pull request by Bupilipili**
### In this branch

- Home page

- [x] When the page loads, the webapp shows the list of categories that could be filtered out by some parameter, for example by the category name.
- [x] Along with the category name, you will display some numeric values per category that come from the API.
- [x] When the user clicks (or taps) on a category item, the application navigates to the details page.

- Details page

- [x] In the details page, the webapp retrieves data from the API to show detailed data in the category.
- [x] When the user clicks on the "Back" button (<), the user navigates to the home page.

- Testing requirements

- [x] Create unit tests for pure functions in your Redux code.
- [x] Create integration tests for your application using the React Testing Library.

- Technical requirements

- [x] The project is a single-page application (SPA) built with React and Redux.
- [x] The data retrieved from the API should be stored in the Redux store.
- [x] You should filter the data that you retrieve from the API using a Filter stateless component.
- [x] Every page (the main page and the pages for each item) should have a unique route within the SPA.
- [x] The project should be deployed and accessible online using [Render](https://render.com/), [Netlify](https://www.netlify.com/) or [GitHub Pages](https://pages.github.com/).
- [x] Add a link to the online version of your app to your README file.

### CODE REVIEW

File: src/components/Home.css [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+* {
+  font-family: 'Open Sans', sans-serif;
+}
+
+.container {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  text-align: center;
+}
```
Comment (VICTOR-CHUKWU) [inline src/components/Home.css:19]: - [x] Good job in creating a nice app, but we need to stick to the project design and guide, in the project design, the cards are stacked in a two-column grid separated by the different colour of alternate cards and no spaces between each card while your design is stacked in a one-column grid and there are spaces between cards, kindly fix this
your design :point_down: 
Screenshot from 2023-09-08 00-47-59
given design :point_down: 
Screenshot from 2023-04-28 11-04-25

File: README.md [hunk 2]
```diff
@@ -1 +1,137 @@
...
+Open the index.html file on the browser or use a VS code extension server to start the project`
+
+- run ***npm install*** in your terminal to download all the packages needed to make the project run
+- run ***npm run start*** or ***npm start*** to open the project in your browser
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+### Deployment
+
+This app will be deployed on Netlify.
```
Comment (VICTOR-CHUKWU) [inline README.md]: - [x] according to the project requirement, kindly add the link to your deployed app

File: src/App.css [hunk 3]
```diff
@@ -0,0 +1,3 @@
+body {
+  display: block;
+}
```
Comment (VICTOR-CHUKWU) [inline src/App.css]: - [x] you already styled your body in your index.css, so kindly remove this file to make your code base cleaner

File: src/components/Details.css [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  height: 24px;
+  animation: spin 1s linear infinite;
+  position: relative;
+  left: 48%;
+}
+
+@keyframes spin {
+  0% { transform: rotate(0deg); }
+  100% { transform: rotate(360deg); }
+}
```
Comment (VICTOR-CHUKWU) [inline src/components/Details.css:45]: - [x] Good job, the loading spinner looks great :+1:

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @Bupilipili , 

## STATUS: CHANGES REQUIRED :recycle: 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Comment by Bupilipili** [comment]
Thanks for the review, am working on the changes 😃 .

File: src/components/Details.css (hunk 4 above)
Comment (Bupilipili) [reply src/components/Details.css:45]: Appreciated 💯 .

**Review by danmainah — APPROVED** [review]
Hi @Bupilipili , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️  Implemented the change requested added a link to the deployed demo and ensured the demo was working correctly.
✔️ Implemented the change requested to redesign the homepage and ensure the layout matches the given design.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Bupilipili** [comment]
Thanks for the review 💯

**Pull request merged**